substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/Target_jdk_jfr_internal_JVM.java

```diff
@@ -300,19 +373,25 @@ public static boolean getAllowedToDoEventRetransforms() {
     /** See {@link JVM#createJFR}. */
     @Substitute
     private static boolean createJFR(boolean simulateFailure) throws IllegalStateException {
+        if (!HasJfrSupport.get()) {
+            throw jfrNotSupportedException();
+        }
         return SubstrateJVM.get().createJFR(simulateFailure);
     }
 
     /** See {@link JVM#destroyJFR}. */
     @Substitute
     private static boolean destroyJFR() {
+        if (!HasJfrSupport.get()) {
+            throw jfrNotSupportedException();
+        }
         return SubstrateJVM.get().destroyJFR();
     }
 
     /** See {@link JVM#isAvailable}. */
     @Substitute
     public static boolean isAvailable() {
-        return true;
+        return HasJfrSupport.get();
```

Comment:
It looks like the JDK code expects this to `throw` instead of actually return a `boolean`: https://github.com/openjdk/jdk/blob/jdk-25%2B22/src/jdk.jfr/share/classes/jdk/jfr/internal/JVMSupport.java#L70   I suspect that may be a workaround for what Native Image did previously. 

I see that you recompute `JVMSupport.notAvailable` with `JfrNotAvailableTransformer` to work around this. 

Do you think it's worth changing the `JVMSupport#checkAvailability` JDK code to actually read the return value of `isAvailable()` so we don't need the double workaround?